### ISSUE TITLE
Add LightGBM-Based Cross-Sectional Alpha Model (Milestone Case Study)

### ISSUE THREAD

**Issue by christophermoverton** [christophermoverton/stratlake-trade-engine#188]
##  **Issue Title**

Add LightGBM-Based Cross-Sectional Alpha Model (Milestone Case Study)

---

##  **Labels**

* enhancement
* ml
* alpha-model
* research
* milestone-15
* case-study

---

### **Objective**

Implement a new machine learning–based alpha model using **LightGBM** to expand the alpha catalog and serve as a second ML case study alongside the existing XGBoost implementation.

This model will integrate fully into the deterministic alpha pipeline, enabling evaluation, candidate selection, and portfolio allocation workflows.

---

### **Context**

The system currently supports:

```
Features → Alpha → Predict → Align → Validate → Evaluate → Aggregate → 
Persist → Register → Compare → Candidate Selection → Allocation → Portfolio
```

An XGBoost-based alpha model has already been implemented as a baseline ML case study.
To improve robustness and demonstrate model diversity in candidate selection, we should introduce a **LightGBM-based alternative**.

LightGBM offers:

* faster training on large cross-sectional datasets
* improved handling of sparse features
* different inductive bias vs XGBoost (useful for diversification)

This addition is especially important for:

* **correlation filtering (Milestone 15)**
* **candidate diversity**
* **selection robustness**

---

#### 1. New Alpha Model Implementation

Create a new model module:

```
src/research/alpha/lightgbm_model.py
```

Responsibilities:

* conform to `BaseAlphaModel` interface
* deterministic training + prediction
* no data leakage (strict temporal handling)
* reproducible outputs

---

#### 2. Feature Integration

* reuse existing feature sets (momentum, volatility, trend, liquidity)
* ensure strict column validation
* maintain deterministic ordering (`symbol`, `ts_utc`)

---

#### 3. Training Workflow

* time-aware training window (half-open intervals)
* fixed hyperparameters (no randomness unless seeded)
* deterministic LightGBM config

---

#### 4. Prediction Output

* cross-sectional predictions per timestamp
* compatible with existing evaluation pipeline

---

#### 5. Alpha Catalog Registration

Update:

```
src/research/alpha/builtins.py
src/research/alpha/catalog.py
```

Add entry:

```
ml_cross_sectional_lgbm_<version>
```

---

#### 6. Config Integration

Update config:

```
configs/alphas_2026_q1.yml
```

Add a working example configuration for the LightGBM alpha.

---

#### 7. CLI Compatibility

Ensure compatibility with:

```
src/cli/run_alpha_evaluation.py
src/cli/compare_alpha.py
```

No special-case logic required.

---

#### 8. Candidate Selection Compatibility

Validate that the model:

* flows through candidate selection (Milestone 15)
* participates in:

  * eligibility filtering
  * correlation pruning
  * allocation

---

#### 9. Example Workflow

Add/update example:

```
docs/examples/lightgbm_alpha_case_study.py
```

Demonstrate:

```
train → predict → evaluate → register → select → portfolio
```

---

### **Acceptance Criteria**

* [ ] LightGBM model implemented and conforms to `BaseAlphaModel`
* [ ] Deterministic training and prediction verified
* [ ] No leakage (passes strict validation)
* [ ] Successfully runs via `run_alpha_evaluation`
* [ ] Produces valid alpha artifacts:

  * `ic_timeseries.csv`
  * `alpha_metrics.json`
* [ ] Appears in alpha comparison CLI
* [ ] Successfully participates in candidate selection pipeline
* [ ] Included in at least one candidate-driven portfolio run
* [ ] Example script executes end-to-end successfully

---

### **Suggested Release Content**

* Introduces a second ML alpha model (LightGBM)
* Expands alpha catalog for candidate selection diversity
* Strengthens correlation filtering and allocation robustness
* Demonstrates end-to-end ML → portfolio workflow under governance

---

### **Notes**

* Keep hyperparameters fixed for reproducibility (no tuning in this issue)
* Avoid introducing new dependencies beyond LightGBM
* Ensure strict parity with XGBoost implementation structure
* Future milestone can introduce:

  * hyperparameter tuning
  * ensemble models
  * model stacking

---

### **Outcome**

A fully integrated LightGBM alpha model that:

* increases alpha diversity
* improves candidate selection realism
* validates the robustness of Milestone 15 governance layer
